### DISCUSSION TITLE
How about basket trading strategy

### DISCUSSION THREAD

**Discussion by wangzhe3224** [nautechsystems/nautilus_trader#529] [Q&A] [answered]
Hi,

Here is some background why I ask this question: in equities trading, there is a set of popular strategy tends to trading a basket of signal names as a whole. for example, statistical arbitrage which could run intraday at different frequency (depending on how fast the computation can be done after sampling and how fast can we trade). Usually the basket could contain from 2 to 200 or even more names. as I mentioned, this kind of strategy tends to trade at some sample times, as the computation could be heavy and execution could be slow.

I noticed that the project run backtest and trading based on single names (Instruments), correct me if I am wrong please. I understand that you can always involves multiple instruments in 1 strategy. however, we need some kind of sync to make sure all the instrument in the basket get the sample time and data, and produce a target basket, them try to execute them as a whole. Like partial fill is not great as it breaks strategies to some degree.  

I guess this probably more a client side thing to implement. But I am wondering do you have any suggestion to construct this kind of strategy in nautulus trader? or even build in some function in to core engine, or you think it is better just leave it to the client side.

thank you
Zhe

**Comment by yohplala** [comment]
Hi @wangzhe3224 ,
I will unfortunately not be of great help as I have not started yet to deep dive in this direction but, if my understanding of your ticket is correct, it is a goal for me.
'Once upon a time', I raised related questions in #324, to which @ian-wazowski & @cjdsellers tried to kindly answer by providing directions.
This ticket is about how setting up a rebalancing strategy, i.e. rebalancing amount of cryptos, which altogether make a 'basket' as you name it (if my understanding is correct once again). At the moment, I am working on the brick in charge of defining the target shares for each, ...since some time now... and expect at some point to be done... probably this summer 2022? :)
So if you make any progress on this, I will be terribly curious about it. And if not, this is something I should be able to help you with in some months! :)
Bests,

**Comment by cjdsellers** [comment, marked as answer]
Hi Zhe,

You're right that you'd want a single strategy to handle a rebalancing from one portfolio state to a target state.
Assuming you have those { 2 .. 200 } instruments available, then you could just fetch those with the `self.cache.instrument(...)` method (if you needed to reference them for price and quantity).

Then at this stage you would need to write your own logic to execute towards your target portfolio state. Something as simple as calculating the differences (within a threshold or not), and submitting the necessary orders.

This rebalancing approach is very popular though, and will likely be one of the first execution algos introduced, however for the time being it falls into the category of specific strategy logic, and so must be written by the trader.

Hope this helps to shed some light!
